nuclia/sdk/resource.py

```diff
@@ -47,6 +51,61 @@ def create(*args, **kwargs) -> str:
         rid = resource.uuid
         return rid
 
+    @kb
+    @pretty
+    def ask(
+        self,
+        *,
+        rid: Optional[str] = None,
+        slug: Optional[str] = None,
+        query: Union[str, dict, AskRequest],
+        answer_json_schema: Optional[Dict[str, Any]] = None,
+        answer_json_file: Optional[str] = None,
+        filters: Optional[Union[List[str], List[Filter]]] = None,
+        **kwargs,
+    ):
+        ndb: NucliaDBClient = kwargs["ndb"]
+
+        if answer_json_file is not None:
+            if os.path.exists(answer_json_file):
+                with open(answer_json_file, "r") as json_file_handler:
+                    answer_json_schema = json.load(json_file_handler)
+
+        if isinstance(query, str):
+            req = AskRequest(
+                query=query,
+                answer_json_schema=answer_json_schema,
+            )
+            if filters is not None:
+                req.filters = filters
+        elif isinstance(query, dict):
+            try:
+                req = AskRequest.model_validate(query)
+            except ValidationError:
+                raise ValueError("Invalid AskRequest object.")
+
+        elif isinstance(query, AskRequest):
+            req = query
+        else:
+            raise ValueError("Invalid query type. Must be str, dict or AskRequest.")
```

Comment:
some unit tests would be nice on this param parsing. and Maybe add some more error handling for the loading from file path